src/Tkmm/Localization/LocaleManager.cs

```diff
@@ -40,6 +46,35 @@ private LocaleManager(LocalesJson json)
         };
     }
 
+    private static string GetSystemDefaultLanguage()
+    {
+        try {
+            // Get the current culture and map it to a supported language
+            string currentCulture = System.Globalization.CultureInfo.CurrentUICulture.Name;
+            string langCode = currentCulture.Split('-')[0].ToLowerInvariant();
+            string countryCode = currentCulture.Contains("-") ? currentCulture.Split('-')[1].ToUpperInvariant() : "";
+            
+            return langCode switch
+            {
+                "en" => countryCode == "GB" ? "en_GB" : "en_US",
+                "ja" => "ja_JP",
+                "fr" => countryCode == "CA" ? "fr_CA" : "fr_FR",
+                "es" => countryCode == "MX" ? "es_MX" : "es_ES",
+                "de" => "de_DE",
+                "nl" => "nl_NL",
+                "it" => "it_IT",
+                "ru" => "ru_RU",
+                "ko" => "ko_KR",
+                "zh" => countryCode == "TW" ? "zh_TW" : "zh_CN",
+                _ => "en_US"  // Default to English if no matching language
+            };
+        }
+        catch {
```

Comment:
If errors are known to occur, it should be fixed - otherwise it should be logged